src/engine.rs

```diff
@@ -771,18 +707,18 @@ impl Reedline {
             if (column, row) == (0, 0) {
                 (0, 0)
             } else if row + 1 == terminal_size.1 {
-                self.stdout.queue(Print("\r\n\r\n"))?.flush()?;
+                self.painter.paint_carrige_return()?;
```

Comment:
I am unsure about this interface. Any recommendations?

Reply:
I think that's fine